neqo-transport/src/connection.rs

```diff
@@ -600,15 +600,14 @@ impl Connection {
     }
 
     /// Set or clear the qlog for this connection.
-    pub fn set_qlog(&mut self, qlog: Option<NeqoQlog>) {
-        let conn_ql = Rc::new(RefCell::new(qlog));
-        self.loss_recovery.set_qlog(conn_ql.clone());
-        self.qlog = conn_ql;
+    pub fn set_qlog(&mut self, qlog: &NeqoQlog) {
+        self.loss_recovery.set_qlog(qlog);
+        self.qlog = qlog.clone();
     }
 
     /// Get the qlog (if any) for this connection.
-    pub fn qlog_mut(&mut self) -> Rc<RefCell<Option<NeqoQlog>>> {
-        self.qlog.clone()
+    pub fn qlog_mut(&mut self) -> &mut NeqoQlog {
```

Comment:
We don't need to hand out a mutable reference here, do we?

Reply:
Related to this reasoning above: https://github.com/mozilla/neqo/pull/776#discussion_r449216284